torch/utils/data/dataset.py

```diff
@@ -293,19 +295,40 @@ def __len__(self):
         return len(self.indices)
 
 
-def random_split(dataset: Dataset[T], lengths: Sequence[int],
+def random_split(dataset: Dataset[T], lengths_or_frac: Sequence[Union[int, float]],
                  generator: Optional[Generator] = default_generator) -> List[Subset[T]]:
     r"""
     Randomly split a dataset into non-overlapping new datasets of given lengths.
+    If a list of fractions is given, the lengths will be computed automatically as floor(frac * len(dataset)).
+    If the sum of the list of the fractions is not equal to 1 or the split leaves a remainder,
+    a new entry will be appended
+    to the end of the list with the remaining fraction
+
     Optionally fix the generator for reproducible results, e.g.:
 
     >>> random_split(range(10), [3, 7], generator=torch.Generator().manual_seed(42))
+    >>> random_split(range(30), [0.3, 0.3, 0.4], generator=torch.Generator().manual_seed(42))
 
     Args:
         dataset (Dataset): Dataset to be split
-        lengths (sequence): lengths of splits to be produced
+        lengths_or_frac (sequence): lengths or fractions of splits to be produced
         generator (Generator): Generator used for the random permutation.
     """
+    if math.isclose(sum(lengths_or_frac), 1):
```

Comment:
Why do we need to use `isclose`?

Reply:
It is possible that the fraction sometimes does not actually sum up to 1 maybe due to a floating point error (ie 0.99999) so isclose should handle these cases

Reply:
There is a corner case that `sum` is larger than 1. With a large dataset, the `remainder` can be a negative value.